python/CMakeLists.txt

```diff
@@ -305,6 +314,9 @@ if (PYARROW_BUNDLE_ARROW_CPP)
   if (MSVC)
     bundle_arrow_implib(ARROW_SHARED_IMP_LIB)
     bundle_arrow_implib(ARROW_PYTHON_SHARED_IMP_LIB)
+    if (ARROW_GPU_FOUND)
+      bundle_arrow_implib(ARROW_GPU_SHARED_IMP_LIB)
+    endif()
```

Comment:
We probably want to make a CMake list of libraries to bundle, then iterate over that with `foreach`

Reply:
I believe fixing bundle logic is not a part of this PR (I cannot test it myself anyway).